scarlet/interpolation.py

```diff
@@ -295,6 +293,26 @@ def get_separable_kernel(dy, dx, kernel=lanczos, **kwargs):
     kyx = np.outer(ky, kx)
     return kyx, y_window, x_window
 
+def sinc_interp(coord_hr, coord_lr, Fm):
+    '''
+    INPUTS:
+    ------
+        coord_hr: Coordinates of the high resolution grid
+        coord_lr: Coordinates of the low resolution grid
+        Fm: Sample at positions coord_lr
+    OUTPUTS:
+    -------
+        result:  interpolated  samples at positions coord_hr
+    '''
+    a, b = coord_hr
+    A, B = coord_lr
+    hx = np.abs(A[1]-A[0])
+    hy = np.abs(B[np.int(np.sqrt(np.size(B)))+1] - B[0])
+
+    assert hx != 0
+
+    return np.array([Fm[k] * sinc2D((a-A[k])/(hx),(b-B[k])/(hy)) for k in range(len(A))]).sum(axis=0)
```

Comment:
Can you avoid the for loop by writing `sinc2D` to operate on arrays?

Reply:
The problem is not the sinc. a is an array. It's the dimensions of a and A that do not match and made me use the loop.

Reply:
Once again, as I say this, I realise there is a smart solution that uses newaxis to shunt the for loop. It works, but I want to confirm that tomorrow on my new machine where I am perfectly sure of the setup.

Reply:
newaxis trick works